apis/release/v1beta1/types.go

```diff
@@ -27,9 +27,10 @@ import (
 
 // A ChartSpec defines the chart spec for a Release
 type ChartSpec struct {
-	Repository    string               `json:"repository"`
-	Name          string               `json:"name"`
-	Version       string               `json:"version"`
+	Repository    string               `json:"repository,omitempty"`
+	Name          string               `json:"name,omitempty"`
+	Version       string               `json:"version,omitempty"`
+	URL           string               `json:"url,omitempty"`
```

Comment:
Might be worth adding comments on these to indicate when fields are required (e.g. if not using URL you need to provide Repository / Name / Version).